src/components/Plots/ObservationHistoryPlot.vue

```diff
@@ -0,0 +1,288 @@
+              className: className,
+              start: previousObservation.start,
+              end: previousObservation.end,
+              toggle: 'tooltip',
+              html: true,
+              type: 'range'
+            });
+            index++;
+            previousObservationIndex = i + 1;
+          }
+          previousObservation = observation;
+        }
+        visGroups.add({ id: index, content: previousObservationIndex });
+        visData.add({
+          id: index,
+          group: index,
+          observationId: previousObservation.id,
+          title:
+            'telescope: ' +
+            previousObservation.site +
+            '.' +
+            previousObservation.enclosure +
+            '.' +
+            previousObservation.telescope +
+            previousObservation.percent_completed +
+            previousObservation.fail_reason +
+            '<br/>start: ' +
+            previousObservation.start.replace('T', ' ') +
+            '<br/>end: ' +
+            previousObservation.end.replace('T', ' '),
+          className: 'timeline_observation ' + previousObservation.state,
+          start: previousObservation.start,
+          end: previousObservation.end,
+          toggle: 'tooltip',
+          html: true,
+          type: 'range'
+        });
+        index++;
+        timeline_min = new Date(visData.get(index - 1)['start']);
+        timeline_max = new Date(visData.get(index - 1)['end']);
+        if (index > 12) {
+          for (let k = index - 1; k >= index - 12; k--) {
+            let start = new Date(visData.get(k)['start']);
+            let end = new Date(visData.get(k)['end']);
+            if (start < timeline_min) {
+              timeline_min = start;
+            }
+            if (end > timeline_max) {
+              timeline_max = end;
+            }
+          }
+          timeline_max.setMinutes(timeline_max.getMinutes() + 30);
+          timeline_min.setMinutes(timeline_min.getMinutes() - 30);
+        }
+      }
+      return { datasets: visData, groups: visGroups, window: { start: timeline_min, end: timeline_max } };
+    }
+  },
+  watch: {
+    data: function() {
+      let datasets = this.toVis;
+      // Need to first zero out the items and groups or vis.js throws an error
+      this.plot.setItems(new vis.DataSet());
+      this.plot.setGroups(new vis.DataSet());
+      this.plot.setGroups(datasets.groups);
+      this.plot.setItems(datasets.datasets);
+      this.plot.setWindow(datasets.window.start, datasets.window.end);
+    }
+  },
+  mounted: function() {
+    this.plot = this.buildPlot();
+    let that = this;
+    this.plot.on('click', function(event) {
+      if (event.item !== null) {
+        // An observation on the timeline was cliked, get that observation info
+        let item = that.toVis.datasets.get(event.item);
+        if (item !== null) {
+          that.$emit('observationClicked', item.observationId);
+        }
+      }
+    });
```

Comment:
You could remove line 227 by using an arrow function:
```
this.plot.on('click', event => {
      if (event.item !== null) {
        // An observation on the timeline was cliked, get that observation info
        let item = this.toVis.datasets.get(event.item);
        if (item !== null) {
          this.$emit('observationClicked', item.observationId);
        }
      }
    });
```

Reply:
Nice, done